src/components/text-field/react/TextField.tsx

```diff
@@ -81,35 +87,106 @@ const COMPONENT_NAME = `${COMPONENT_PREFIX}TextField`;
  */
 const CLASSNAME: string = getRootClassName(COMPONENT_NAME);
 
+/**
+ * The minimum number of rows that we want to display on the text area
+ */
+const MIN_ROWS = 2;
+
 /**
  * The default value of props.
  */
 const DEFAULT_PROPS: Partial<TextFieldProps> = {
     hasError: false,
     isDisabled: false,
     isValid: false,
+    minimumRows: MIN_ROWS,
     theme: Theme.light,
     type: TextFieldType.input,
 };
-
+/**
+ * Hook that allows to calculate the number of rows needed for a text area.
+ * @param minimumNumberOfRows Minimum number of rows that we want to display.
+ * @return rows to be used and a callback to recalculate
+ */
+const useComputeNumberOfRows = (
+    minimumNumberOfRows: number | undefined = MIN_ROWS,
```

Comment:
It's probably easier to read as `minimumNumberOfRows?: number` rather than `minimumNumberOfRows: number | undefined`
Or alternativaly, we could ask the `minimumNumberOfRows` to be always provided in htis hook and use the default prop value make sure it's not `undefined` (we also would not need the `MIN_ROWS` const anymore)

Reply:
For the second option, you have to tell TypeScript that the `minimumNumberOfRows` cannot be `undefined` like this:
`minimumNumberOfRows = DEFAULT_PROPS.minimumNumberOfRows as number`

Reply:
For me, it feels more natural to avoid passing the prop and let the component figure out what to use. I will make the changes so it reads as `optional`